Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyViewModel.kt

```diff
@@ -17,125 +17,156 @@
 package com.example.compose.jetsurvey.survey
 
 import android.net.Uri
-import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
 import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.setValue
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
-import androidx.lifecycle.viewModelScope
-import com.example.compose.jetsurvey.util.getDefaultDateInMillis
-import kotlinx.coroutines.launch
+import com.example.compose.jetsurvey.survey.question.Superhero
 
 const val simpleDateFormatPattern = "EEE, MMM d"
 
 class SurveyViewModel(
-    private val surveyRepository: SurveyRepository,
     private val photoUriManager: PhotoUriManager
 ) : ViewModel() {
 
-    private val _uiState = MutableLiveData<SurveyState>()
-    val uiState: LiveData<SurveyState>
-        get() = _uiState
+    private val questionOrder: List<SurveyQuestion> = listOf(
+        SurveyQuestion.FREE_TIME,
+        SurveyQuestion.SUPERHERO,
+        SurveyQuestion.LAST_TAKEAWAY,
+        SurveyQuestion.FEELING_ABOUT_SELFIES,
+        SurveyQuestion.TAKE_SELFIE
+    )
 
-    var askForPermissions by mutableStateOf(true)
-        private set
-
-    private lateinit var surveyInitialState: SurveyState
+    private var questionIndex = 0
 
     // Uri used to save photos taken with the camera
     private var uri: Uri? = null
 
+    // ----- Responses exposed as State -----
+
+    private val _freeTimeResponse = mutableStateListOf<Int>()
+    val freeTimeResponse: List<Int>
+        get() = _freeTimeResponse
+
+    private val _superheroResponse = mutableStateOf<Superhero?>(null)
+    val superheroResponse: Superhero?
+        get() = _superheroResponse.value
+
+    private val _takeawayResponse = mutableStateOf<Long?>(null)
+    val takeawayResponse: Long?
+        get() = _takeawayResponse.value
+
+    private val _feelingAboutSelfiesResponse = mutableStateOf<Float?>(null)
+    val feelingAboutSelfiesResponse: Float?
+        get() = _feelingAboutSelfiesResponse.value
+
+    private val _selfieUriResponse = mutableStateOf<Uri?>(null)
+    val selfieUriResponse: Uri?
+        get() = _selfieUriResponse.value
+
+    // ----- Survey status exposed as LiveData -----
```

Comment:
I'm surprised to see a `LiveData` here. Is there a valid reason to use `LiveData` vs a Compose `State`? If so, consider adding a larger comment explaining why you're using the two different versions in this sample.

Reply:
I was originally thinking it would be good to show that either can be used, but now I think that it's just more confusing that way. I'm not sure a comment would really make that more clear, so I switched them to State also.